pyatlan/client/credential.py

```diff
@@ -46,6 +48,32 @@ def get(self, guid: str) -> CredentialResponse:
         if not isinstance(raw_json, dict):
             return raw_json
         return CredentialResponse(**raw_json)
+    
+    @validate_arguments
+    def get_all(self, filter: dict, limit: int = None, offset: int = None) -> CredentialResponseList:
+        """
+        Retrieves all credentials based on the provided filter and optional pagination parameters.
+
+        :param filter: A dictionary specifying the filter criteria (required).
+        :param limit: Maximum number of credentials to retrieve (optional).
+        :param offset: Number of credentials to skip before starting retrieval (optional).
+        :returns: A CredentialResponseList instance.
+        :raises: AtlanError on any error during API invocation.
+         """
+        params = {"filter": filter}
```

Comment:
I would also make the `filter` query param `optional `(let's allow the user to decide if they want to use it or not):

```suggestion
   if filter is not None:
            params["filter"] = filter
```